### PR TITLE
support ipv6 in e2e policy tests

### PR DESCRIPTION
**Pull request by aojea**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind deprecation
> /kind design
> /kind documentation

/kind failing-test

> /kind feature
> /kind flake

**What this PR does / why we need it**:

Network policy tests are not testing the feature correctly for IPv6

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/network/network_policy.go [hunk 1]
```diff
@@ -1450,9 +1453,15 @@ var _ = SIGDescribe("NetworkPolicy [LinuxOnly]", func() {
...
 			}
 
-			podServerAllowCIDR := fmt.Sprintf("%s/24", podServerStatus.Status.PodIP)
+			allowMask := 24
+			exceptMask := 32
+			if utilnet.IsIPv6String(podServerStatus.Status.PodIP) {
+				allowMask = 64
+				exceptMask = 128
+			}
+			podServerAllowCIDR := fmt.Sprintf("%s/%d", podServerStatus.Status.PodIP, allowMask)
```
Comment (danwinship) [inline test/e2e/network/network_policy.go:1462]: oops, we're testing undefined behavior anyway; we're saying to allow to, eg, "192.168.1.5/24" when we should be saying to allow to "192.168.1.**0**/24".

You need to do:

    _, podServerAllowSubnet, err := net.ParseCIDR(fmt.Sprintf("%s/%d", podServerStatus.Status.PodIP, allowMask))
    framework.ExpectNoError(err, "could not parse allow subnet")
    podServerAllowCIDR := podServerAllowSubnet.String()

likewise in the test below

File: test/e2e/network/network_policy.go [hunk 2]
```diff
@@ -1515,8 +1524,14 @@ var _ = SIGDescribe("NetworkPolicy [LinuxOnly]", func() {
...
+			podServerIP := fmt.Sprintf("%s/%d", podServerStatus.Status.PodIP, exceptMask)
 			// Exclude podServer's IP with an Except clause
 			podServerExceptList := []string{podServerIP}
```
Comment (danwinship) [inline test/e2e/network/network_policy.go:1536]: It's not really an "IP" when it has a prefix length. Maybe just get rid of the `podServerIP` variable and do this like in the above test

Comment (aojea) [reply test/e2e/network/network_policy.go:1536]: yep

File: test/e2e/network/network_policy.go (hunk 1 above)
Comment (aojea) [reply test/e2e/network/network_policy.go:1462]: yeah, strictly speaking that's correct, just curiosity, do you think that app should "normalize it" and use it or reject it?

Comment (danwinship) [reply test/e2e/network/network_policy.go]: There's no way for us to know whether the user actually meant "192.168.1.5/**32**" or "192.168.1.**0**/24"; either mistake is possible. So we can't "normalize it". Ideally we would have rejected this at validation time.

File: test/e2e/network/network_policy.go [hunk 3]
```diff
@@ -1373,8 +1374,13 @@ var _ = SIGDescribe("NetworkPolicy [LinuxOnly]", func() {
...
 			}
-
-			podServerCIDR := fmt.Sprintf("%s/32", podServerStatus.Status.PodIP)
+			mask := 32
+			if utilnet.IsIPv6String(podServerStatus.Status.PodIP) {
+				mask = 128
+			}
+			_, podServerSubnet, err := net.ParseCIDR(fmt.Sprintf("%s/%d", podServerStatus.Status.PodIP, mask))
+			framework.ExpectNoError(err, "could not parse pod subnet")
+			podServerCIDR := podServerSubnet.String()
```
Comment (danwinship) [inline test/e2e/network/network_policy.go:1383]: you don't need to do this in the 32/128 case; in that case we're keeping all the bits of the pod IP, so just `podServerCIDR := fmt.Sprintf("%s/%d", podServerStatus.Status.PodIP, mask)` is fine

**Comment by aojea** [comment]
@danwinship I totally forget about this one, addressed the comments
PTAL

**Pull request merged**